### PR TITLE
feat(Typography): update font size and line heights

### PR DESCRIPTION
**Pull request by teimurjan**
[FX-2142](https://toptal-core.atlassian.net/browse/FX-2142)

### Description

- Updated `font-size` and `line-height` according to BASE 2.0.
- Refactored styles.
- Moved visual tests to Picasso.

### How to test

- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Comment by teimurjan** [comment]
> Can you recheck image snapshots of Breadcrumbs, Modal? For some reason, in the image snapshots, it looks broken, but not in the storybook.

Both breadcrumbs and Modal snapshots are fine. The font is different a bit because we're not using the Toptal font family for visuals. What is exactly broken?

**Comment by teimurjan** [comment]
> ShowMore image snapshots show that the icon moved to a side, is it intentional?

Yes, previously it had an invalid DOM structure (`Link -> Typography`). However, we recommend wrapping `Link` into `Typography` to change the `font-size`.

**Comment by teimurjan** [comment]
> The Sidebar doesn't look good, but I have looked into the design and it's not updated to the correct line-height. Maybe talk with the base team and create a follow-up task?

Sure, I'll do it. But can you clarify how exactly it doesn't look good? It looks good to me.

File: packages/picasso/src/ShowMore/ShowMore.tsx [hunk 1]
```diff
@@ -60,25 +60,25 @@ export const ShowMore = forwardRef<HTMLSpanElement, Props>(function ShowMore(
...
-          onClick={() => {
-            setShownMore(!shownMore)
-            onToggle()
-          }}
-          className={classes.toggleText}
-          underline='none'
-        >
-          <Typography size='medium' color='blue'>
+        <Typography size='medium'>
+          <Link
```
Comment (denieler) [inline packages/picasso/src/ShowMore/ShowMore.tsx:64]: isn't it a super breaking change? if we need to wrap now Link inside Typography, but not vice-versa? this breaking change will be very difficult to spot and fix in projects

File: packages/picasso/src/Typography/styles.ts [hunk 2]
```diff
@@ -1,77 +1,71 @@
...
 
-PicassoProvider.override(() => ({
+import { toMuiVariant } from './utils'
+
+PicassoProvider.override(({ palette, typography }) => ({
+  // If typography variant has a unique mui variant mapping, you should override styles here
   MuiTypography: {
-    h1: {
-      lineHeight: '1.5em'
+    [toMuiVariant('heading', 'xlarge')]: {
```
Comment (denieler) [inline packages/picasso/src/Typography/styles.ts:9]: here we override mui styles and we need to know which style we are overriding, but after introducing `toMuiVariant` function we are telling here which Picasso variants we are overriding in MUI styles, which kinda doesn't make sense to me

File: packages/picasso-lab/src/OverviewBlock/story/Multiline.example.tsx [hunk 3]
```diff
@@ -35,12 +35,12 @@ const MultilineExample = () => (
         <OverviewBlock
           value='$5,758,716.46'
           label='Pending receipt'
-          variant='label-blue'
+          variant='label-green'
```
[the remark this answers is not in the document]
Comment (teimurjan) [reply packages/picasso-lab/src/OverviewBlock/story/Multiline.example.tsx:38]: I'll check with Milos.

File: packages/picasso/src/Typography/Typography.tsx [hunk 4]
```diff
@@ -34,7 +28,7 @@ export interface Props
   /** Text align of the inner text */
   align?: PropTypes.Alignment
   /** Size of the inner text */
-  size?: SizeType<'small' | 'medium' | 'large' | 'xlarge'> | 'inherit'
+  size?: SizeType
```
[the remark this answers is not in the document]
Comment (teimurjan) [reply packages/picasso/src/Typography/Typography.tsx:31]: I need to use `Typography`'s `SizeType` in `utils`. I don't have any other choice.

PS: Importing directly from the `Typography` file introduces a circular dep and breaks the tests.

[the remark this answers is not in the document]
Comment (denieler) [reply packages/picasso/src/Typography/Typography.tsx:31]: maybe we can duplicate this type for utils? (i don't know the best answer for this, but feel this change makes code readability slightly worse)

File: packages/picasso/src/Typography/styles.ts (hunk 2 above)
Comment (teimurjan) [reply packages/picasso/src/Typography/styles.ts:11]: Discussed on the daily meeting. A spike ticket was created and mentioned in the code.

File: packages/picasso/src/ShowMore/ShowMore.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso/src/ShowMore/ShowMore.tsx:64]: no-no, it's 👍 this is a better combination of Link and Typography, right? so it's the necessary fix

**Review by denieler — APPROVED** [review]
💪 🙇

**Pull request merged**